### PR TITLE
Simplify TestCase.assertExpected()

### LINKED ISSUE #2814: Simplify logic on TestCase.assertExpected()
## 🐛 Bug

My IDE spotted a tiny bug/readability issue on the TestCase.assertExpected() method:
https://github.com/pytorch/vision/blob/fdca30739b01a6dfb732f23289701d0e9860d39f/test/common_utils.py#L153-L159

The method `TestCase.assertEqual()` does not return a value, hence if successful the variable `equal` will be `None`. The code still work but it can be simplified to improve readability:
```
            try:
                self.assertEqual(output, expected, prec=prec)
            except Exception:
                return accept_output("updated output")
```

### PR DESCRIPTION
**Pull request by datumbox**
Minor Refactoring on `TestCase.assertExpected()` to fix #2814

### CODE REVIEW

File: test/common_utils.py [hunk 1]
```diff
@@ -150,12 +150,9 @@ def accept_output(update_type):
...
 
         if ACCEPT:
-            equal = False
             try:
-                equal = self.assertEqual(output, expected, prec=prec)
+                self.assertEqual(output, expected, prec=prec)
             except Exception:
-                equal = False
-            if not equal:
                 return accept_output("updated output")
```
Comment (vfdev-5) [inline test/common_utils.py:156]: I have an impression that `return accept_output("updated output")` is always returned ?

Comment (datumbox) [reply test/common_utils.py:156]: @vfdev-5 Absolutely, the return can go away. Also the `accept_output()` does not return either. Let me make a correction.

Comment (vfdev-5) [reply test/common_utils.py:156]: @datumbox thinking more about this test, I think your code is correct with return in `except` branch. I think the intention is to update the output if `output != expected`.

Comment (datumbox) [reply test/common_utils.py]: @vfdev-5 I saw you approved but could you please have another look? I made some more changes.

**Review by vfdev-5 — APPROVED** [review]
LGTM!

**Comment by vfdev-5** [comment]
Seems like the CI is a bit confused by the last commit as circle ci tests are not shown...

**Comment by fmassa** [comment]
Same thing is happening on https://github.com/pytorch/vision/pull/2812, maybe it's a CircleCI issue?

**Review by fmassa — APPROVED** [review]
Thanks @datumbox !

**Pull request merged**